src/context/AdditionalFormOptionsContext/types.ts

```diff
@@ -1,3 +1,9 @@
-export interface AdditionalFormOptionsContextValue {
+import { Dispatch, SetStateAction } from 'react';
+
+export interface AdditionalFormState {
   readonly?: boolean;
 }
+
+export interface AdditionalFormOptionsContextValue extends AdditionalFormState {
+  setFormState: Dispatch<SetStateAction<AdditionalFormState>>;
```

Comment:
let's name this `updateFormState` and let it receive the form state, and not do direct react setters